### PR TITLE
Hide the trail in the SERP preview if no URL can be generated

### LINKED ISSUE #6860: Unable to generate a content URL for the SERP widget, please provide the url_callback
### Affected version(s)

5.3.0-RC3 

### Description

If you want to reset the redirect target in a news item to "Use default", the following error occurs.

```
LogicException:
Unable to generate a content URL for the SERP widget, please provide the url_callback.

  at vendor/contao/core-bundle/contao/widgets/SerpPreview.php:171
  at Contao\SerpPreview->getUrl(object(NewsModel))
     (vendor/contao/core-bundle/contao/widgets/SerpPreview.php:53)
  at Contao\SerpPreview->generate()
     (vendor/contao/core-bundle/contao/library/Contao/Widget.php:647)
  at Contao\Widget->generateWithError(true)
     (vendor/contao/core-bundle/contao/templates/backend/be_widget.html5:3)
  at include('/home/…/public_html/…/vendor/contao/core-bundle/contao/templates/backend/be_widget.html5')
     (vendor/contao/core-bundle/contao/library/Contao/TemplateInheritance.php:109)
  at Contao\Widget->inherit()
     (vendor/contao/core-bundle/contao/library/Contao/Widget.php:595)
  at Contao\Widget->parse()
     (vendor/contao/core-bundle/contao/classes/DataContainer.php:777)
  at Contao\DataContainer->row()
     (vendor/contao/core-bundle/contao/drivers/DC_Table.php:2199)
  at Contao\DC_Table->edit()
     (vendor/contao/core-bundle/contao/classes/Backend.php:548)
  at Contao\Backend->getBackendModule('news', null)
     (vendor/contao/core-bundle/contao/controllers/BackendMain.php:146)
  at Contao\BackendMain->run()
     (vendor/contao/core-bundle/src/Controller/BackendController.php:44)
  at Contao\CoreBundle\Controller\BackendController->mainAction()
     (vendor/symfony/http-kernel/HttpKernel.php:181)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:76)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:197)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (public/index.php:44)
```

### PR DESCRIPTION
**Pull request by leofeyer**
Fixes #6860

### CODE REVIEW

**Review by aschempp — CHANGES_REQUESTED** [review]
I don't think we should change the exception. Changing it to a routing exception would no longer show the message (to a developer) that a `url_callback` should be added.

Does the `$chunks` part fix the original issue?

**Comment by leofeyer** [comment]
The `$chunks` part fixes the issue when you switch from `external` back to `default`. The exception fixes the issue when you switch from `page` or `article` back to `default`.

**Review by fritzmg — COMMENTED** [review]
We already catch the `RouteParametersException` and the (current) [fix](https://github.com/contao/contao/pull/6854) for https://github.com/contao/contao/issues/6797 would change the exception to a `RouteParametersException` and thus would be fixed by that PR.

If we agree that https://github.com/contao/contao/pull/6854 is the correct fix for https://github.com/contao/contao/issues/6797 then I can continue working on https://github.com/contao/contao/pull/6854 for the other bundles and this PR can be closed.

**Comment by leofeyer** [comment]
This is how the SERP preview now looks if the URL requires an item:



---

This is how the SERP preview now looks if the URL cannot be generated:



---

And this is how the SEPR preview normally looks:

File: core-bundle/contao/widgets/SerpPreview.php [hunk 1]
```diff
@@ -46,28 +47,37 @@ public function generate()
...
-		// Use the base URL for the index page
-		if ($model instanceof PageModel && $alias == 'index')
+		catch (ExceptionInterface)
 		{
-			$url = $trail;
+			// Ignore other routing exceptions
 		}
-		else
+
+		if ($url)
```
Comment (aschempp) [inline core-bundle/contao/widgets/SerpPreview.php]: you could actually move this to the `try` condition, then we wouldn't need the `if ($url)`

**Review by aschempp — DISMISSED** [review]
Appart from a possible improvement 🙃

File: core-bundle/contao/widgets/SerpPreview.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/contao/widgets/SerpPreview.php]: Yes, I thought about that, too. But isn‘t it better if the `try` block only contains the one method call that actually might throw the exception?

Comment (aschempp) [reply core-bundle/contao/widgets/SerpPreview.php]: I don't think it really matters… 🤷

Comment (leofeyer) [reply core-bundle/contao/widgets/SerpPreview.php]: Changed in 5b5de6d69953eb2200130cf61293a49d71b9e07e then.

File: core-bundle/contao/widgets/SerpPreview.php [hunk 2]
```diff
@@ -49,25 +50,30 @@ public function generate()
 
 		try
 		{
-			// Get the URL with a %s placeholder for the alias or ID
+			// Get the URL with an %s placeholder for the alias or ID
```
Comment (fritzmg) [inline core-bundle/contao/widgets/SerpPreview.php]:
```suggestion
			// Get the URL with a %s placeholder for the alias or ID
```

Comment (leofeyer) [reply core-bundle/contao/widgets/SerpPreview.php]: PhpStorm made that change. It probably interprets it as _an s placeholder_.

Comment (fritzmg) [reply core-bundle/contao/widgets/SerpPreview.php]: When read out loud you would still say "with a percent-s placeholder", as 's' alone is not the placeholder we are looking for, right? Would PhpStorm still complain if you put it in quotes? i.e. "with a '%s' placeholder"

Comment (leofeyer) [reply core-bundle/contao/widgets/SerpPreview.php]: Yes, it would. I consider this a bug in PhpStorm.

**Pull request merged**